ethsigner/commandline/src/test/java/tech/pegasys/ethsigner/CommandlineParserClientTlsOptionsTest.java

```diff
@@ -0,0 +1,191 @@
+  private CommandlineParser parser;
+  private NullSignerSubCommand subCommand;
+  private String defaultUsageText;
+
+  @BeforeEach
+  void setup() {
+    subCommand = new NullSignerSubCommand();
+    config = new EthSignerBaseCommand();
+    parser = new CommandlineParser(config, outPrintStream);
+    parser.registerSigner(subCommand);
+
+    final CommandLine commandLine = new CommandLine(new EthSignerBaseCommand());
+    commandLine.addSubcommand(subCommand.getCommandName(), subCommand);
+    defaultUsageText = commandLine.getUsageMessage();
+  }
+
+  @Test
+  void cmdLineIsValidIfOnlyDownstreamTlsIsEnabled() {
+    final String cmdLine =
+        removeFieldFrom(
+            validBaseCommandOptions(),
+            "downstream-http-tls-keystore-file",
+            "downstream-http-tls-keystore-password-file",
+            "downstream-http-tls-invalidate-ca-signed",
+            "downstream-http-tls-known-servers-file");
+
+    final boolean result =
+        parser.parseCommandLine((cmdLine + subCommand.getCommandName()).split(" "));
+
+    assertThat(result).as("CLI Parse result").isTrue();
+    final Optional<ClientTlsOptions> optionalDownstreamTlsOptions = config.getClientTlsOptions();
+    assertThat(optionalDownstreamTlsOptions.isPresent()).as("Downstream TLS Options").isTrue();
+
+    assertThat(optionalDownstreamTlsOptions.get().isTlsEnabled()).as("TLS Enabled").isTrue();
+    assertThat(optionalDownstreamTlsOptions.get().getClientTlsTrustOptions().isEmpty()).isTrue();
+    assertThat(optionalDownstreamTlsOptions.get().getClientTlsCertificateOptions().isEmpty())
+        .isTrue();
+  }
+
+  @Test
+  void cmdLineIsValidWithAllTlsOptions() {
+    final String cmdLine = validBaseCommandOptions();
+
+    final boolean result =
+        parser.parseCommandLine((cmdLine + subCommand.getCommandName()).split(" "));
+
+    assertThat(result).as("CLI Parse result").isTrue();
+    final Optional<ClientTlsOptions> optionalDownstreamTlsOptions = config.getClientTlsOptions();
+    assertThat(optionalDownstreamTlsOptions.isPresent()).as("Downstream TLS Options").isTrue();
+
+    final ClientTlsOptions clientTlsOptions = optionalDownstreamTlsOptions.get();
+    assertThat(clientTlsOptions.isTlsEnabled()).as("TLS Enabled").isTrue();
+    assertThat(clientTlsOptions.getClientTlsTrustOptions().isPresent()).isTrue();
+    final ClientTlsTrustOptions clientTlsTrustOptions =
+        clientTlsOptions.getClientTlsTrustOptions().get();
+    assertThat(clientTlsTrustOptions.getKnownServerFile().get()).isEqualTo(Path.of("./test.txt"));
+    assertThat(clientTlsTrustOptions.isCaSignedServerCertificateAllowed()).isFalse();
+
+    final ClientTlsCertificateOptions clientTlsCertificateOptions =
+        clientTlsOptions.getClientTlsCertificateOptions().get();
+    assertThat(clientTlsCertificateOptions.getKeyStoreFile()).isEqualTo(Path.of("./test.ks"));
+  }
+
+  @Test
+  void cmdLineFailsIfDownstreamKeystoreIsUsedWithoutTlsEnabled() {
```

Comment:
This test isn't showing that the command line fails and it won't until we upgrade to pico cli 4.2. Is there anything that can be tested with this option atm? Otherwise this test can probably go.

Reply:
I left this unit test in because it will fail when we upgrade to picocli 4.2, forcing us to fix it. It was attempting to test the logging on standard out but I took the code out because it resulted in upgrading picocli deprecated methods. I am inclined towards leaving it in unless you are strongly inclining towards removing it.

Reply:
disabled.